### PR TITLE
[WIPTEST] apply  various fixers from python-modernize to ease python3 migration

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
currently unapplied fixes (due to many false positives)

```

lib2to3.fixes.fix_idioms
lib2to3.fixes.fix_set_literal
lib2to3.fixes.fix_tuple_params
lib2to3.fixes.fix_types
libmodernize.fixes.fix_basestring
libmodernize.fixes.fix_classic_division
libmodernize.fixes.fix_dict_six
libmodernize.fixes.fix_file
libmodernize.fixes.fix_filter
libmodernize.fixes.fix_map
libmodernize.fixes.fix_open
libmodernize.fixes.fix_print


libmodernize.fixes.fix_unicode_type
libmodernize.fixes.fix_zip
```

### CODE REVIEW

**Comment by mshriver** [comment]
I broke your branch with a template_upload_ec2 refactor, sorry ;)

File: cfme/fixtures/pytest_selenium.py [hunk 1]
```diff
@@ -46,6 +47,7 @@
 from utils.deprecation import removed_in_fw30
 
 from threading import local
+from functools import reduce
```
Comment (izapolsk) [inline cfme/fixtures/pytest_selenium.py]: what is it for ?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
closing will reopen with a different iteration stategy

**Pull request closed without merging**